mycroft/skills/intent_services/adapt_service.py

```diff
@@ -227,31 +244,63 @@ def take_best(intent, utt):
 
     def register_vocab(self, start_concept, end_concept, alias_of, regex_str):
         """Register vocabulary."""
-        if regex_str:
-            self.engine.register_regex_entity(regex_str)
-        else:
-            self.engine.register_entity(
-                start_concept, end_concept, alias_of=alias_of)
+        with self.lock:
+            if regex_str:
+                self.engine.register_regex_entity(regex_str)
+            else:
+                self.engine.register_entity(
+                    start_concept, end_concept, alias_of=alias_of)
```

Comment:
This is beyond the scope of this PR but I just went down a small rabbit hole of what start vs end concepts were and I think that these need to be renamed to better match it's usage eg Adapt now seems to use `entity_value, entity_type`.

Reply:
i always need to double check what these mean, i approve the renaming :+1: